Classes/git/PBGitRevSpecifier.m

```diff
@@ -14,26 +14,26 @@ @implementation PBGitRevSpecifier
 @synthesize parameters, description, workingDirectory;
 @synthesize isSimpleRef;
 
+// I believe this relates loosely to parts of git-check-ref-format.
+// cf. https://www.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html
+//
+NS_INLINE BOOL ContainsComplexRefCharSequence(NSString *refString)
+{
+	return ([refString hasPrefix:@"-"] ||
+			[refString rangeOfCharacterFromSet:[NSCharacterSet characterSetWithCharactersInString:@" ~^:"]].location != NSNotFound ||
+			[refString rangeOfString:@".."].location != NSNotFound ||
+			[refString rangeOfString:@"@{"].location != NSNotFound);
+}
 
 // internal designated init
 - (id) initWithParameters:(NSArray *)params description:(NSString *)descrip
 {
-    self = [super init];
-    if (!self) return nil;
+	self = [super init];
+	if (!self) return nil;
+
 	parameters = params;
 	description = descrip;
-
-	if (([parameters count] > 1) || ([parameters count] == 0))
-		isSimpleRef =  NO;
-	else {
-		NSString *param = [parameters objectAtIndex:0];
-		if ([param hasPrefix:@"-"] ||
-			[param rangeOfCharacterFromSet:[NSCharacterSet characterSetWithCharactersInString:@"^@{}~:"]].location != NSNotFound ||
-			[param rangeOfString:@".."].location != NSNotFound)
-			isSimpleRef =  NO;
-		else
-			isSimpleRef =  YES;
-	}
+	isSimpleRef = (parameters.count == 1) && !ContainsComplexRefCharSequence(parameters[0]);
```

Comment:
Some good points, particularly the `NSParameterAssert` suggestion, which I had considered, and will do. I wrote this in 2013, and was trying to get it past Rowan at the time (ha!), which meant I tried to keep the change minimal. FWIW, ObjC nil-messaging is baked into my brain by now, for better or worse.

To your points, 

* in my code, `params == nil` would set  `isSimpleRef` to `NO` which is a possibly breaking change from the original code, and also kinda nonsensical.
* Pretty sure `parameters[0]` can only be called if `parameters.count == 1`
* `_Nonnull`ing all the things is a good idea, but more complicated, so I'm deferring.
* I've used this profitably on my own builds for some time now, so I'm pretty sure this always gets called with non-nil `params`.